From 1428ac919f84268c7504f61f4516b540e3ad7a64 Mon Sep 17 00:00:00 2001
From: jinghua <xigu@marvell.com>
Date: Fri, 14 Apr 2017 09:52:09 +0800
Subject: [PATCH 0975/1345] dts: a3700: usb: move from usb static host mode to
 otg mode

commit  61bc8e4bce80c729f2e6881d8ce1cb34750e3dbd from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- This patch changes A3700 devel board dts file to move from USB
  static host mode to OTG mode.

Change-Id: I58717874c1b740d34523ef18a24021a8cac284a2
Signed-off-by: jinghua <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38677
Reviewed-by: Hua Jing <jinghua@marvell.com>
Tested-by: Hua Jing <jinghua@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38757
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts |   19 +++++++++----------
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi   |    2 +-
 2 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 294bc52..e5e7916 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -68,15 +68,9 @@
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 		enable-active-high;
-		regulator-always-on;
 		gpio = <&gpio_exp 1 GPIO_ACTIVE_HIGH>;
 	};
 
-	usb3_phy: usb3-phy {
-		compatible = "usb-nop-xceiv";
-		vcc-supply = <&exp_usb3_vbus>;
-	};
-
 	soc {
 		internal-regs@d0000000 {
 			vcc_sd_reg1: regulator@0 {
@@ -95,13 +89,13 @@
 			};
 
 			u3d@50000 {
-				status = "disabled";
+				status = "okay";
 				phys = <&a3700_comphy 1 COMPHY_USB3>;
 				phy-names = "usb";
 			};
 
 			udc@54100 {
-				status = "disabled";
+				status = "okay";
 				phys = <&utmi_usb32>;
 				phy-names = "usb";
 			};
@@ -125,6 +119,11 @@
 			utmi_usb2: utmi@5f000 {
 				status = "okay";
 			};
+
+			usb_phy: usb_phy@5d800 {
+				vcc-supply = <&exp_usb3_vbus>;
+				status = "okay";
+			};
 		};
 
 		reg_sata: pwr-sata {
@@ -134,7 +133,7 @@
 			regulator-max-microvolt = <12000000>;
 			enable-active-high;
 			regulator-boot-on;
-			gpio = <&expander0 2 GPIO_ACTIVE_HIGH>;
+			gpio = <&gpio_exp 2 GPIO_ACTIVE_HIGH>;
 		};
 
 		reg_5v_sata: v5-sata {
@@ -270,7 +269,7 @@
 /* CON31 */
 &usb3 {
 	status = "okay";
-	usb-phy = <&usb3_phy>;
+	usb-phy = <&usb_phy>;
 	phys = <&utmi_usb32>, /* utmi usb32 dedicated phy */
        <&a3700_comphy 1 COMPHY_USB3>; /* usb3 comphy */
 	phy-names = "usb2", "usb3";
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index f095cf5..14836dc 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -401,7 +401,7 @@
 				status = "disabled";
 			};
 			usb3: usb@58000 {
-				compatible = "marvell,armada3700-xhci",
+				compatible = "marvell,armada-3700-xhci",
 				"generic-xhci";
 				reg = <0x58000 0x4000>;
 				interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
-- 
1.7.9.5

